src/inmanta/data/__init__.py

```diff
@@ -4815,6 +4815,293 @@ def get_compliance_status(self) -> Optional[state.Compliance]:
+        :param updated_resources: A list of resources to insert.
+            On a full compile, this list should contain all resources present in the version.
+            On a partial compile, this list should contain all resources belonging to a resource set that was changed.
+        :param updated_resource_sets: A list of resource sets that were changed.
+        :param base_version: This is the version which the partial compile is based on. None if we are doing a full compile.
+        :param resource_set_names_not_to_bump: These are the resource set names from the base version .
+            which we do not wish to bump. Either because these sets were removed in this partial compile or because
+            these sets received changes. Not applicable for a full compile.
+        :param connection: The connection to use.
+        """
+
+        is_partial_update = base_version is not None
+        resource_sets_to_bump = (
+            """
+            UNION SELECT rscm.resource_set_id
+            FROM public.resource_set_configuration_model AS rscm
+            INNER JOIN public.resource_set AS rs
+                ON rscm.environment=rs.environment
+                AND rscm.resource_set_id=rs.id
+            WHERE rscm.environment=$1
+                AND rscm.model=$12
+                AND NOT EXISTS (
+                    -- extra hoop to make comparison with null names --
+                      SELECT 1
+                      FROM UNNEST($13::text[]) AS outdated_resource_set_names(name)
+                      WHERE outdated_resource_set_names.name IS NOT DISTINCT FROM rs.name
+                    )
+        """
+            if is_partial_update
+            else ""
+        )
+
+        resource_ids = []
+        agents = []
+        attributes = []
+        attribute_hashes = []
+        resource_types = []
+        resource_id_values = []
+        is_undefined = []
+        resource_sets = []
+
+        for r in updated_resources:
+            assert r.environment == environment
+            assert r.model == target_version
+            resource_ids.append(r.resource_id)
+            agents.append(r.agent)
+            attributes.append(json.dumps(r.attributes))
+            attribute_hashes.append(util.make_attribute_hash(r.resource_id, r.attributes))
+            resource_types.append(r.resource_type)
+            resource_id_values.append(r.resource_id_value)
+            is_undefined.append(r.is_undefined)
+            resource_sets.append(r.resource_set)
+
+        values = [
+            environment,
+            target_version,
+            updated_resource_sets,
+            resource_ids,
+            agents,
+            attributes,
+            attribute_hashes,
+            resource_types,
+            resource_id_values,
+            is_undefined,
+            resource_sets,
+        ]
+        if is_partial_update:
+            values.extend([base_version, resource_set_names_not_to_bump])
+
+        query = f"""
```

Comment:
What is the reason for making it one query?

Reply:
The main benefit of having it all in one query is not having to set the resource set uuids in python and managing them.
In this query we have to manage the resource sets that we inserted and link them to the correct resources and configuration models.